content/docs/start/model-management/model-registry.md

```diff
@@ -0,0 +1,289 @@
+model registry actions programmatically. You can learn more about the details of
+GTO in its [documentation](/docs/gto).
+
+We recommend most users to interact with the model registry using the Studio GUI
+as it is easier to use and also has some exclusive features such as the ability
+to view models from multiple git repositories in one place.
+
+</details>
+
+## Versioning models
+
+Now that we have our first model in the model registry, we can start registering
+model versions for the model. We do it by choosing a specific commit in our
+model development history and attaching a version to it to keep an easier track
+of it. We will do that directly in the Studio UI as follows
+
+![Registering model versions](/img/mr-register-model-version.gif)
+
+Since we just saved our model to dvc and added it to the model registry in the
+previous commit, we can just keep the commit which was selected automatically.
+We will also keep the suggested version number. For more details and other ways
+of registering model versions you can have a look at the corresponding
+[documentation](/doc/studio/user-guide/model-registry/register-version).
+
+Once we register our first model version, the model registry will also
+automatically connect it to experiment tracking and all metrics which are
+tracked for each model version will also show up in the model registry. We can
+even explore the experiment directly by clicking on the "Open in Project" button
+on the model detail page.
+
+<details>
+
+#### 💡 Expand to see how registering models works under the hood
+
+Registering the model version as we just did using DVC Studio is equivalent to
+the following GTO command
+
+```
+gto register pool-segmentation [ref] --version v1.0.0
+```
+
+Here, `[ref]` is the git reference/hash we selected from the menu in Studio.
+
+For more details you can have a look at the
+[gto register command reference](doc/gto/command-reference/register).
+
+</details>
+
+## Assigning lifecycle stages
+
+We have a first version for our model and now it is a good time to assign a
+model lifecycle stage to it. You can create any number of lifecycle stages with
+any names you wish but in this example we will only create two stages called
+"dev" and "test".
+
+Stages are created whenever a model version is assigned to them. We will now
+assign our model to the "dev" stage as follows.
+
+![Assigning model stages](/img/mr-assign-model-stage.gif)
+
+<details id="under-the-hood-assigning-model-stages">
+
+#### 💡 Expand to see how assigning model stages works under the hood
+
+Assigning the "dev" stage to the model as we just did using DVC Studio is
+equivalent to the following GTO command
+
+```
+gto assign pool-segmentation --version v1.0.0 --stage dev
+```
+
+For more details you can have a look at the
+[gto assign command reference](doc/gto/command-reference/assign).
+
+</details>
+
+## Changing (and removing) stage assignments
```

Comment:
Are we focusing on the wrong workflow here? I think it's more likely that users register a new version and assign that to `dev` rather than need to deprecate and have nothing in `dev`.

Reply:
Hmm, I guess you're right. We can include manually deprecating a stage in the link to the user guide we already have there anyway.

Reply:
Thinking about this a bit more - adding an extra model version and replacing it introduced a lot of additional steps. It would populate the model history with more events which is nice, but iw would be repeating steps that were done already - another experiment run (and ideally with different parameters which also needs to be addressed), another model version registration, another stage assignment...only to note that when you assign a different version to "dev", it in fact replaces the old version in dev.

I don't think it adds much knowledge, but it does add a lot of extra steps. I'd perhaps just add the model to the "prod" stage, include that teaser of the next chapter (we'll see that we can use that to automate stuff) and then only mention that if you add a new version and assign it to a stage, it will remove any prior assignments to that stage (by default).